src/Core/Content/Product/ProductException.php

```diff
@@ -102,4 +103,20 @@ public static function reviewNotActive(): self|ReviewNotActiveExeption
             'Reviews not activated'
         );
     }
+
+    /**
+     * @deprecated tag:v6.8.0 - reason:return-type-change - Will only return `self` in the future
+     */
+    public static function originalIdNotFound(string $originalId): self|\RuntimeException
+    {
+        if (!Feature::isActive('v6.8.0.0')) {
+            return new \RuntimeException(\sprintf('Cannot find originalId %s in listing mapping', $originalId));
+        }
+
+        return new self(
+            Response::HTTP_BAD_REQUEST,
+            self::PRODUCT_ORIGINAL_ID_NOT_FOUND,
+            \sprintf('Cannot find originalId %s in listing mapping', $originalId)
```

Comment:
please use the offered string replacement functionality. see examples above. pass `originalId` as parameter